packages/angular/src/generators/ng-add/utilities/e2e-project.migrator.ts

```diff
@@ -68,39 +70,77 @@ export class E2eProjectMigrator extends ProjectMigrator {
   }
 
   validate(): ValidationResult {
-    // TODO: implement validation when multiple apps are supported
+    if (!this.project) {
+      return null;
+    }
+
+    if (this.isProtractorE2eProject()) {
+      if (
+        this.tree.exists(
+          this.projectConfig.targets.e2e.options.protractorConfig
+        )
+      ) {
+        return;
+      }
+
+      return (
+        `An e2e project with Protractor was found but "${this.projectConfig.targets.e2e.options.protractorConfig}" could not be found.\n` +
+        `Make sure the "${this.appName}.architect.e2e.options.protractorConfig" is valid or the "${this.appName}" project is removed from "angular.json".`
+      );
+    } else if (this.isCypressE2eProject()) {
+      const configFile = this.getCypressConfigFile();
+      if (configFile && !this.tree.exists(configFile)) {
+        return `An e2e project with Cypress was found but "${configFile}" could not be found.`;
```

Comment:
Similar to the Karma one, I kept this as it was before, I just moved it. Unless you think otherwise, I would prefer addressing this on the PR adding support for multiple apps and keep this PR focused on ESLint changes.